### PR TITLE
Refs #23919 -- Removed obsolete mentions of unicode.

### PR DESCRIPTION
**Pull request by vytisb**

### CODE REVIEW

**Comment by timgraham** [comment]
You might also look for "unicode" ... some things should stay as is, but, for example, I think it's appropriate to remove "unicode" in a case like `contrib/auth/management/__init__.py:    :returns: The username as a unicode string, or an empty string if the` considering that "string" in Python 3 means "unicode string".

**Comment by vytisb** [comment]
@timgraham I have removed a bunch of `unicode` mentions as you suggested. I have left `unicode` or `unicode string` in some places where it references a string containing non-ASCII characters or where the surrounding code seems outdated (such as [here](https://github.com/django/django/blob/8377a98ca5a9d51f6cf705d75276cb0380fffad6/django/utils/dateformat.py#L88-L90)).

**Comment by timgraham** [comment]
@claudep, would you like to double check this?

**Comment by claudep** [comment]
Yes, sure, I will.

File: docs/ref/unicode.txt [hunk 1]
```diff
@@ -48,7 +47,7 @@ General string handling
 
 Whenever you use strings with Django -- e.g., in database lookups, template
 rendering or anywhere else -- you have two choices for encoding those strings.
-You can use normal Unicode strings or bytestrings (starting with a 'b').
+You can use normal strings or bytestrings (starting with a 'b').
```
Comment (claudep) [inline docs/ref/unicode.txt]: I think this section needs some deeper rewriting. I don't think we will continue to defensively code expecting possible bytesstrings in most of the framework. This has performance costs for almost no gain. However, this could be done in a separate PR later.

File: docs/ref/unicode.txt [hunk 2]
```diff
@@ -74,13 +73,13 @@ using your application -- and if that person chooses a different setting, your
 code must still continue to work. Ergo, it cannot rely on that setting.
 
 In most cases when Django is dealing with strings, it will convert them to
-Unicode strings before doing anything else. So, as a general rule, if you pass
-in a bytestring, be prepared to receive a Unicode string back in the result.
+strings before doing anything else. So, as a general rule, if you pass
```
Comment (claudep) [inline docs/ref/unicode.txt]: This is typically a case where the sentence becomes meaningless. Again, not sure we will solve this in this PR.

**Comment by claudep** [comment]
https://code.djangoproject.com/ticket/27778 is the ticket asking for a rewritten unicode data page.

File: docs/ref/unicode.txt (hunk 1 above)
Comment (aaugustin) [inline docs/ref/unicode.txt]: Yes, this whole page needs rewriting. It isn't just a matter of updating some sentences; the basic premise of the page changes when str == unicode instead of str == bytes.

I created a ticket about that this morning before I noticed this PR. I think it's best to leave out changes to this file from this PR and do them in a dedicated PR.
